api/v1alpha1/osbuildconfig_types.go

```diff
@@ -0,0 +1,133 @@
+
+	// Details defines what to build
+	Details BuildDetails `json:"build_details"`
+	// Triggers defines when to build
+	Triggers BuildTriggers `json:"build_triggers"`
+}
+
+// BuildDetails includes all the information needed to build the image
+type BuildDetails struct {
+	// Distribution is the name of the O/S distribution
+	Distribution string `json:"distribution"`
+	// Customizations defines the changes to be applied on top of the base image (optional)
+	Customizations *Customizations `json:"customizations,omitempty"`
+	// TargetImage defines the requested output image
+	TargetImage TargetImage `json:"target_image"`
+}
+
+// Customizations defines the changes to be applied on top of the base image
+type Customizations struct {
+	// Packages is a list of RPM packages to install (optional)
+	Packages []string `json:"packages,omitempty"`
+	// Users is the list of Users to add to the image (optional)
+	Users []User `json:"users,omitempty"`
+	// Services defines the services to enable or disable (optional)
+	Services *Services `json:"services,omitempty"`
+}
+
+// User defines a single user to be configured
+type User struct {
+	// Name is the username for the new user
+	Name string `json:"name"`
+	// Groups is the groups to add the user to (optional)
+	Groups []string `json:"groups,omitempty"`
+	// Key is the user's SSH public key (optional)
+	Key *string `json:"key,omitempty"`
+}
+
+type Services struct {
+	// Enabled is the list of services to enabled (optional)
+	Enabled []string `json:"enabled,omitempty"`
+	// Disabled is the list of services to disabled (optional)
+	Disabled []string `json:"disabled,omitempty"`
+}
+
+type TargetImage struct {
+	// Architecture defines target architecture of the image
+	Architecture string `json:"architecture"`
+	// ImageType defines the target image type
+	TargetImageType TargetImageType `json:"target_image_type"`
+	// OSTree is the OSTree configuration of the build (optional)
+	OSTree *OSTreeConfig `json:"ostree,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=edge-container;edge-installer
+type TargetImageType string
+
+// OSTreeConfig defines the OSTree ref details
+type OSTreeConfig struct {
+	// Url is the URL of the target build
+	Url *string `json:"url"`
+	// Ref is the ref of the target build
+	Ref *string `json:"ref"`
+	// Parent is the ref of the parent of target build (Optional)
+	Parent *string `json:"parent"`
+}
+
+type BuildTriggers struct {
+	// ConfigChange if True trigger a new build upon any change in this BuildConfig CR (optional)
+	ConfigChange *bool `json:"config_change,omitempty"`
+	// WebHook defines the way to trigger a build using a REST call (optional)
+	WebHook *buildv1.WebHookTrigger `json:"webhook,omitempty"`
+}
+
+// OSBuildConfigStatus defines the observed state of OSBuildConfig
+type OSBuildConfigStatus struct {
+	// LastVersion denotes the number of the last OSBuild CR created for thie OSBuildConfig CR
+	LastVersion *int `json:"lastversion,omitempty"`
```

Comment:
what will it be the use for this field?

Reply:
Again, I'm mimicking the BuildConfig behavior. Since the system will keep old OSBuild CRs, this field will allow the user to find the last OSBuild CR as its name will be `<OSBuildConfig Name>-<LastVersion>`